inc/link.class.php

```diff
@@ -659,13 +669,45 @@ static function rawSearchOptionsToAdd($itemtype = null) {
 
       if (!Session::isCron()
           && !isCommandLine() && isset($_SESSION['glpiactiveentities_string'])) {
-         $newtab['joinparams']['condition'] = getEntitiesRestrictRequest('AND', 'NEWTABLE');
+         if ($itemtype == User::getType()) {
+            // User would need a different query as entities_id is set on the user's profiles
+            $newtab['joinparams']['condition'] = getEntitiesRestrictRequest('AND', 'NEWTABLE');
+         } else {
+            $item_table = $itemtype::getTable();
+            $item = new $itemtype();
+            $is_recurive = $item->maybeRecursive();
+
+            if ($is_recurive) {
+               $newtab['joinparams']['condition'] = "AND ( `NEWTABLE`.`entities_id` = `$item_table`.`entities_id` OR (
```

Comment:
Unless I am wrong, using `getEntitiesRestrictRequest` with `$is_recursive: true` should fix the initial issue.

Reply:
No it's more complex than that (see updated description).